### PR TITLE
Update LLVM/MHLO to week of 2022/11/14

### PR DESCRIPTION
**Pull request by gargaroff**
Green LLVM commit: e864ac694540342d5e59f59c525c5082f2594fb8
Green MHLO commit: eab364ba2a66bd0613efb94f8a738c1c97aaee92

The commits listed above are the green commits from Nov. 14.

Changelog:

mlir-hlo changed their directory structure, requiring two `include_directories` directives to make it buildable.
Generation of `#map` has changed, requiring large NFC changes to test references.

### CODE REVIEW

File: test/mlir/conversion/krnl_to_affine/buffer_copy.mlir [hunk 1]
```diff
@@ -320,14 +320,15 @@ func.func private @copy_to_runtime_start_indices(%p0 : index, %p1 : index) -> ()
   krnl.copy_to_tile_buffer %B, %AA [%p0, %p1], %f0 {padToNext=[1,1]}: memref<4x6xf32>, memref<39x56xf32>
   return
 // mlir2FileCheck.py -n'{"0": "ORGINAL", "1": "BUFFER", "cst": "ZERO"}' -a'["start0", "start1"]'
-// CHECK-DAG: #map = affine_map<()[s0] -> (-s0 + 39, 4)>
-// CHECK-DAG: #map1 = affine_map<()[s0] -> (-s0 + 56, 6)>
+// CHECK-DAG: #map = affine_map<(d0, d1) -> (d0, d1)>
```
Comment (ljfitz) [inline test/mlir/conversion/krnl_to_affine/buffer_copy.mlir:323]: Is this map used in any other lines being checked? If not, why are we checking it?

File: test/mlir/conversion/krnl_to_affine/buffer_copy.mlir [hunk 2]
```diff
@@ -320,14 +320,15 @@ func.func private @copy_to_runtime_start_indices(%p0 : index, %p1 : index) -> ()
...
+// CHECK-DAG: #map1 = affine_map<()[s0] -> (-s0 + 39, 4)>
+// CHECK-DAG: #map2 = affine_map<()[s0] -> (-s0 + 56, 6)>
```
Comment (ljfitz) [inline test/mlir/conversion/krnl_to_affine/buffer_copy.mlir:325]: Would it be sensible to use symbolic match names here?

File: test/mlir/onnx/onnx_lowering_with_canonicalize.mlir [hunk 3]
```diff
@@ -2698,40 +2736,40 @@ func.func @test_cumsum_constant_axis_exclusive_reverse_mode(%arg0: tensor<2x3xf6
...
 // CHECK-DAG:       [[LOOP_0_:%.+]]:2 = krnl.define_loops 2
-// CHECK:           krnl.iterate([[LOOP_0_]]#0, [[LOOP_0_]]#1) with ([[LOOP_0_]]#0 -> [[AXIS_:%.+]] = 0 to 2, [[LOOP_0_]]#1 -> [[I_0_:%.+]] = 0 to 3){
-// CHECK:             [[VAR_4_:%.+]]:2 = krnl.get_induction_var_value([[LOOP_0_]]#0, [[LOOP_0_]]#1) : (!krnl.loop, !krnl.loop) -> (index, index)
-// CHECK:             [[VAR_5_:%.+]] = arith.addi [[VAR_4_]]#1, [[CST_1_]] : index
-// CHECK:             [[VAR_6_:%.+]] = arith.cmpi slt, [[VAR_5_]], [[CST_3_]] : index
-// CHECK:             [[VAR_7_:%.+]] = arith.select [[VAR_6_]], [[VAR_5_]], [[VAR_4_]]#1 : index
-// CHECK:             [[LOAD_INPUT_MEM_:%.+]] = krnl.load [[INPUT_]]{{.}}[[VAR_4_]]#0, [[VAR_7_]]{{.}} : memref<2x3xf64>
-// CHECK:             [[VAR_9_:%.+]] = arith.select [[VAR_6_]], [[LOAD_INPUT_MEM_]], [[CST_0_dot_000000_]] : f64
-// CHECK:             krnl.store [[VAR_9_]], [[RES_1_]]{{.}}[[VAR_4_]]#0, [[VAR_4_]]#1] : memref<2x3xf64>
+// CHECK:           krnl.iterate([[LOOP_0_]]#0, [[LOOP_0_]]#1) with ([[LOOP_0_]]#0 -> [[I_0_:%.+]] = 0 to 2, [[LOOP_0_]]#1 -> [[I_1_:%.+]] = 0 to 3){
```
Comment (ljfitz) [inline test/mlir/onnx/onnx_lowering_with_canonicalize.mlir:2739]: Why the change of variable names? It seems to unnecessarily complicate the review.

**Review by ljfitz — CHANGES_REQUESTED** [review]
There are many ref changes that seem unnecessary, making it very difficult to see what has actually changed.

File: test/mlir/conversion/krnl_to_affine/buffer_copy.mlir [hunk 4]
```diff
@@ -320,14 +320,15 @@ func.func private @copy_to_runtime_start_indices(%p0 : index, %p1 : index) -> ()
   krnl.copy_to_tile_buffer %B, %AA [%p0, %p1], %f0 {padToNext=[1,1]}: memref<4x6xf32>, memref<39x56xf32>
   return
 // mlir2FileCheck.py -n'{"0": "ORGINAL", "1": "BUFFER", "cst": "ZERO"}' -a'["start0", "start1"]'
-// CHECK-DAG: #map = affine_map<()[s0] -> (-s0 + 39, 4)>
-// CHECK-DAG: #map1 = affine_map<()[s0] -> (-s0 + 56, 6)>
-// CHECK-LABEL:  func private @copy_to_runtime_start_indices
+// CHECK-DAG:   [[MAP_0_:#.+]] = affine_map<(d0, d1) -> (d0, d1)>
```
Comment (gargaroff) [inline test/mlir/conversion/krnl_to_affine/buffer_copy.mlir:323]: missed this

File: test/mlir/conversion/krnl_to_affine/buffer_copy.mlir [hunk 5]
```diff
@@ -397,30 +398,30 @@ func.func private @copy_to_runtime_start_indices_larger_source(%p0 : index, %p1
   krnl.copy_to_tile_buffer %B, %AA [%i2, %i5, %p0, %p1], %f0 {padToNext=[3,3]}: memref<4x6xf32>, memref<5x10x39x56xf32>
   return
 // mlir2FileCheck.py -n'{"0": "ORGINAL", "1": "BUFFER", "cst": "ZERO"}' -a'["start0", "start1"]'
-// CHECK-DAG: #map = affine_map<()[s0] -> (-s0 + 39, 4)>
-// CHECK-DAG: #map1 = affine_map<()[s0] -> (-s0 + 56, 6)>
+// CHECK-DAG:   [[MAP_2_:#.+]] = affine_map<()[s0] -> (-s0 + 39, 4)>
+// CHECK-DAG:   [[MAP_3_:#.+]] = affine_map<()[s0] -> (-s0 + 56, 6)>
```
Comment (gargaroff) [inline test/mlir/conversion/krnl_to_affine/buffer_copy.mlir:402]: rename this to match old name

File: test/mlir/conversion/krnl_to_affine/buffer_copy.mlir [hunk 6]
```diff
@@ -472,13 +473,13 @@ func.func @copy_to_nested_partial(%p0 : index, %p1 : index) -> () {
   }
   return
 
-// CHECK-DAG: #map = affine_map<(d0) -> (-d0 + 45, 10)>
+// CHECK-DAG:   [[MAP_1_:#.+]] = affine_map<(d0) -> (-d0 + 45, 10)>
```
Comment (gargaroff) [inline test/mlir/conversion/krnl_to_affine/buffer_copy.mlir:476]: rename

File: test/mlir/conversion/krnl_to_affine/buffer_copy.mlir [hunk 7]
```diff
@@ -644,14 +645,14 @@ func.func private @copy_from_partial_runtime(%p0 : index, %p1 : index) -> () {
 
   return
 // mlir2FileCheck.py -n'{"0": "ORGINAL", "1": "BUFFER", "cst": "ZERO"}' -a'["start0", "start1"]'
-// CHECK-DAG: #map = affine_map<()[s0] -> (-s0 + 39, 4)>
-// CHECK-DAG: #map1 = affine_map<()[s0] -> (-s0 + 56, 6)>
+// CHECK-DAG:   [[MAP_1_:#.+]] = affine_map<()[s0] -> (-s0 + 39, 4)>
+// CHECK-DAG:   [[MAP_2_:#.+]] = affine_map<()[s0] -> (-s0 + 56, 6)>
```
Comment (gargaroff) [inline test/mlir/conversion/krnl_to_affine/buffer_copy.mlir:649]: rename

File: test/mlir/krnl/krnl_memset.mlir [hunk 8]
```diff
@@ -15,7 +15,7 @@ func.func @lowering_krnl_memset(%arg0: memref<1xi64>, %arg1: memref<1xi64>) -> (
   krnl.memset %2, %cst_f1 {delayed = false} : memref<?x?xf16, #map_2ds>
   return %2 : memref<?x?xf16, #map_2ds>
 
-// CHECK-DAG: #map = affine_map<()[s0] -> (s0 ceildiv 64)>
+// CHECK-DAG: [[MAP_2_:#.+]] = affine_map<()[s0] -> (s0 ceildiv 64)>
```
Comment (gargaroff) [inline test/mlir/krnl/krnl_memset.mlir:18]: rename

**Comment by gargaroff** [comment]
Glad you made me do this @ljfitz. It was huge pain in the butt and my fingers hurt from copy-search-select-pasting so much, but it was definitely worth it, looking at the diff now 😄

File: test/mlir/krnl/krnl_enable_memory_pool.mlir [hunk 9]
```diff
@@ -47,7 +47,7 @@ func.func @test_allocs_not_lowered(%arg0: memref<10x10xf32>, %arg1: memref<10x10
     return %0 : memref<10x10xf32>
 }
 
-// CHECK: [[MAP:#.+]] = affine_map<(d0, d1)
+// CHECK: [[MAP:#.+]] = affine_map<(d0, d1) -> (0, d1 floordiv 64, 0, d0 floordiv 32, d0 mod 32, d1 mod 64)
```
Comment (ljfitz) [inline test/mlir/krnl/krnl_enable_memory_pool.mlir:50]: Do you know where this change came from?

File: test/mlir/onnx/onnx_lowering.mlir [hunk 10]
```diff
@@ -197,7 +197,7 @@ func.func private @test_exp(%arg0 : tensor<?x10xf32>) -> tensor<*xf32> {
   // CHECK: [[C0_0:%.+]] = arith.constant 0 : index
   // CHECK: [[C0_1:%.+]] = arith.constant 0 : index
   // CHECK: [[DIM_2:%.+]] = memref.dim %arg0, [[C0_1]] : memref<?x10xf32>
-  // CHECK: krnl.iterate([[DEF_LOOPS]]#0, [[DEF_LOOPS]]#1) with ([[DEF_LOOPS]]#0 -> %arg1 = 0 to #map([[DIM_2]]), [[DEF_LOOPS]]#1 -> %arg2 = 0 to 10){
+  // CHECK: krnl.iterate([[DEF_LOOPS]]#0, [[DEF_LOOPS]]#1) with ([[DEF_LOOPS]]#0 -> %arg1 = 0 to #map1([[DIM_2]]), [[DEF_LOOPS]]#1 -> %arg2 = 0 to 10){
```
Comment (ljfitz) [inline test/mlir/onnx/onnx_lowering.mlir:200]: Was it not sensible to address this file with the `[[MAP:#.+]] =` approach?

Comment (gargaroff) [reply test/mlir/onnx/onnx_lowering.mlir:200]: nope, simply missed this file

**Review by ljfitz — APPROVED** [review]
This looks good for upstreaming, I'd expect a question about the `affine_map<(d0, d1) -> (0, d1 floordiv 64, 0, d0 floordiv 32, d0 mod 32, d1 mod 64)` change, so if you already know where it came from you should add a review comment to explain,

**Comment by ljfitz** [comment]
Closed in favor of https://github.com/onnx/onnx-mlir/pull/ 1905

**Pull request closed without merging**